inst/include/plant/species.h

```diff
@@ -192,6 +195,45 @@ std::vector<double> Species<T,E>::net_reproduction_ratio_by_cohort() const {
   return ret;
 }
 
+template <typename T, typename E>
+std::vector<double> Species<T,E>::integrate_shared_resource_consumption(const E& environment) const {
+  int n_resources = environment.vars.state_size;
+  std::vector<double> ret;
+  ret.reserve(n_resources);
+
+  for(size_t i = 0; i < n_resources; i++){
+    std::vector<double> heights;
+    heights.reserve(size());
+
+    std::vector<double> consumption;
+    consumption.reserve(size());
+    // cohort heights are in descending order - we need ascending for integration
+    for(auto it = cohorts.rbegin(); it != cohorts.rend(); ++it) {
+      heights.push_back(it->height());
+      consumption.push_back(it->compute_consumption(i));
+    }
+
+    // integrate over cohorts; use sum when only a few cohorts
+    if(size() < 3) {
+      ret.push_back(accumulate(consumption.begin(), consumption.end(), 0));
+    } else {
+      ret.push_back(util::trapezium(heights, consumption));
+    }
+  }
+  return ret;
+}
+
+// Temporary - refactor into something more generic
+template <typename T, typename E>
+std::vector<double> Species<T,E>::auxilliary_variable_by_cohort(std::string aux_name) const {
+  std::vector<double> ret;
```

Comment:
I don't think this is used any more but could be a replacement pattern for other `variable_by_cohort` methods.